stan/math/prim/prob/exp_mod_normal_lccdf.hpp

```diff
@@ -48,69 +48,65 @@ return_type_t<T_y, T_loc, T_scale, T_inv_scale> exp_mod_normal_lccdf(
   scalar_seq_view<T_inv_scale> lambda_vec(lambda);
   size_t N = max_size(y, mu, sigma, lambda);
 
-  for (size_t n = 0; n < N; n++) {
+  for (size_t n = 0, size_y = size(y); n < size_y; n++) {
     if (is_inf(y_vec[n])) {
-      if (y_vec[n] > 0.0) {
-        return ops_partials.build(negative_infinity());
-      } else {
-        return ops_partials.build(0.0);
-      }
+      return y_vec[n] > 0.0 ? ops_partials.build(negative_infinity())
+                            : ops_partials.build(0.0);
     }
+  }
 
+  for (size_t n = 0; n < N; n++) {
     const T_partials_return y_dbl = value_of(y_vec[n]);
     const T_partials_return mu_dbl = value_of(mu_vec[n]);
     const T_partials_return sigma_dbl = value_of(sigma_vec[n]);
     const T_partials_return lambda_dbl = value_of(lambda_vec[n]);
     const T_partials_return u = lambda_dbl * (y_dbl - mu_dbl);
     const T_partials_return v = lambda_dbl * sigma_dbl;
-    const T_partials_return v_sq = v * v;
-    const T_partials_return v_over_sqrt_two = v / SQRT_TWO;
+    const T_partials_return v_over_sqrt_two = v * INV_SQRT_TWO;
+    const T_partials_return inv_sigma = 1 / sigma_dbl;
     const T_partials_return scaled_diff
-        = (y_dbl - mu_dbl) / (SQRT_TWO * sigma_dbl);
+        = (y_dbl - mu_dbl) * INV_SQRT_TWO * inv_sigma;
     const T_partials_return scaled_diff_sq = scaled_diff * scaled_diff;
-    const T_partials_return erf_calc1 = 0.5 * (1 + erf(u / (v * SQRT_TWO)));
-    const T_partials_return erf_calc2
+    const T_partials_return erf_calc
         = 0.5 * (1 + erf(u / (v * SQRT_TWO) - v_over_sqrt_two));
+    const T_partials_return exp_term = exp(0.5 * square(v) - u);
 
-    const T_partials_return deriv_1
-        = lambda_dbl * exp(0.5 * v_sq - u) * erf_calc2;
+    const T_partials_return deriv_1 = lambda_dbl * exp_term * erf_calc;
     const T_partials_return deriv_2
-        = SQRT_TWO_OVER_SQRT_PI * 0.5
-          * exp(0.5 * v_sq - square(-scaled_diff + v_over_sqrt_two) - u)
-          / sigma_dbl;
+        = SQRT_TWO_OVER_SQRT_PI * 0.5 * exp_term
+          * exp(-square(-scaled_diff + v_over_sqrt_two)) * inv_sigma;
     const T_partials_return deriv_3
-        = SQRT_TWO_OVER_SQRT_PI * 0.5 * exp(-scaled_diff_sq) / sigma_dbl;
+        = SQRT_TWO_OVER_SQRT_PI * 0.5 * exp(-scaled_diff_sq) * inv_sigma;
 
-    const T_partials_return ccdf_
-        = 1.0 - erf_calc1 + exp(-u + v_sq * 0.5) * (erf_calc2);
+    const T_partials_return ccdf_n
+        = 1.0 - 0.5 * (1 + erf(u / (v * SQRT_TWO))) + exp_term * erf_calc;
```

Comment:
This can be simplified further to remove an addition:

```
const T_partials_return ccdf_n = 0.5 - 0.5 * erf(u / (v * SQRT_TWO)) + exp_term * erf_calc;
```

Reply:
Thanks. For symmetry I've modified also the cdf and lcdf ones: there are no savings in those cases, but it makes it easier to see the similarity of those codes.